pinot-common/src/main/java/com/linkedin/pinot/common/config/ColumnPartitionConfig.java

```diff
@@ -114,4 +114,29 @@ public static String rangesToString(List<IntRange> ranges) {
     }
     return builder.toString();
   }
+
+  @Override
+  public boolean equals(Object o) {
+
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    ColumnPartitionConfig that = (ColumnPartitionConfig) o;
+
+    if (_numPartitions != that._numPartitions) {
+      return false;
+    }
+    return _functionName != null ? _functionName.equals(that._functionName) : that._functionName == null;
```

Comment:
That's right. It was auto-generated code, fixed to remove the unnecessary null check.